### PR TITLE
Implementation and tests - non seekable multipart form files

### PR DESCRIPTION
**Pull request by ordinaryorange**
Refer #1398

### CODE REVIEW

File: src/Net/Http.fs [hunk 1]
```diff
@@ -1327,9 +1327,9 @@ module internal HttpHelpers =
                     else read
 
             override x.CanRead = true
-            override x.CanSeek = false
+            override x.CanSeek = match length with | None -> false | Some _ -> true
             override x.CanWrite = false
-            override x.Length with get () = length
+            override x.Length with get () = length |> Option.defaultWith (fun () -> NotSupportedException() |> raise)
```
Comment (cartermp) [inline src/Net/Http.fs:1332]: What do you think about a more descriptive message like in `Stream`? https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=net-6.0#exceptions

Comment (ordinaryorange) [reply src/Net/Http.fs:1332]: If flipped on this a couple of times. But now rethinking, I reckon a simple `failwith "One or more of the encompassed streams are not seekable and the length cannot be determined"`. We can probably do away with the `NotSupportedException` type as the `CombinedStream` is only internal to `FSharp.data`

**Review by cartermp — APPROVED** [review]
Thank you! Will cut a release likely beginning of next week.

**Comment by cartermp** [comment]
Should be live in NuGet promptly

**Pull request merged**